base/strings/basic.jl

```diff
@@ -610,3 +610,52 @@ function filter(f, s::AbstractString)
     end
     String(take!(out))
 end
+
+## string first and last ##
+
+"""
+    first(str::AbstractString, nchar::Integer)
+
+Get a string consisting of the first `nchar` characters of `str`.
+
+```jldoctest
+julia> first("∀ϵ≠0: ϵ²>0", 0)
+""
+
+julia> first("∀ϵ≠0: ϵ²>0", 1)
+"∀"
+
+julia> first("∀ϵ≠0: ϵ²>0", 3)
+"∀ϵ≠"
+```
+"""
+function first(str::AbstractString, nchar::Integer)
+    if 0 <= nchar <= 1
+        return str[1:nchar]
+    end
+    str[1:nextind(str, 1, nchar-1)]
+end
```

Comment:
Should it return a `SubString`?

Reply:
My recommendation in #23765 is that not. Usually `first` and `last` would be used with a small `nchar` so there is not much gain from `SubString` (or actually a loss if original string is very large, we get only a small part of it and the original string can be discarded). Observe that it can be a `SubString` if the original string is a `SubString`.